packages/dashboard-core-plugins/src/useIrisGridModel.ts

```diff
@@ -8,10 +8,12 @@ export type IrisGridModelFetch = () => Promise<dh.Table>;
 export type IrisGridModelFetchErrorResult = {
   error: NonNullable<unknown>;
   status: 'error';
+  model: undefined;
 };
 
 export type IrisGridModelFetchLoadingResult = {
   status: 'loading';
+  model: undefined;
```

Comment:
I don't like this - now if you use either of these specific types, it will suggest `.model` even though it's always `undefined` for this specific type.

![image](https://github.com/user-attachments/assets/34a55231-22a1-46b2-88ec-a6a342575006)

Instead, I'd narrow the type first where you were doing this, e.g.
```typescript
// Currently:
const { model } = fetchResult;

// Instead do 
  const model =
    fetchResult.status === 'success' ? fetchResult.model : undefined;
```

I think that's more clear. You could also make it a type predicate, or a function to get model from the result, but adding the `model: undefined` here pollutes the types IMO